### DISCUSSION TITLE
Styles not applied to buttons

### DISCUSSION THREAD

**Discussion by Faro1991** [AvaloniaUI/Avalonia#9130] [Q&A] [answered]
Hi everyone,

maybe this is really simple, but I have two buttons that lose their styling once I try to bind functions to them. The relevant piece of code is the following (imo):

```xml
<StackPanel>
  <StackPanel.Styles>
  <Style Selector="Button">
    <Setter Property="Background" Value="#EE4D2E"/>
    <Setter Property="Foreground" Value="#F2F5F4"/>
  <!--Some more styles-->
</StackPanel.Styles>

    
<!--Some other elements-->
<Button x:Name="LoginButton" HorizontalAlignment="Stretch" Command="{Binding PerformLogin}">Login!</Button>
<Button x:Name="CloseButton" HorizontalAlignment="Stretch">Close!</Button>
</StackPanel>
```

This results in the following render:
![image](https://user-images.githubusercontent.com/6747919/194710334-066e3870-d8b1-4e70-823c-3e6920356f33.png)
(the red border outlines the `LoginButton` element, as it is _really_ hard to see without the style applied)

Anyone able to tell, what I might be missing here? To my understanding, the `Command` property should not break the style selector.

**Comment by timunie** [comment, marked as answer]
If the command is null or the command doesn't allow execution, the button will become disabled. That seems to be the case here to me.

**Reply by maxkatz6** [reply]
Yes, disabled button styles are overriding normal state. Also check styles troubleshooting documentation page.

**Reply by Faro1991** [reply]
Thanks for the info. I already went through the Troubleshooting page, but maybe there's something I missed on that one.

**Reply by timunie** [reply]
Im sure your Command is either not found or disabled for some reason.

**Reply by Faro1991** [reply]
It was not found, yes. I misunderstood the View/ViewModel relationship in terms of what to put into my Binding here. Somehow I assumed that I could just name the function and be done. Putting
```xml
<Button x:Name="LoginButton" HorizontalAlignment="Stretch" Command="{Binding $parent[ControlElement].DataContext.PerformLogin}">Login!</Button>
```
did the trick. Thanks!
